server/plugin.go

```diff
@@ -426,6 +428,36 @@ func (p *Plugin) OnActivate() error {
 		}
 	}
 
+	remoteID, err := p.API.RegisterPluginForSharedChannels(model.RegisterPluginOpts{
```

Comment:
It is important to notice that I'm registering everything here, and adding the hooks, and subscribing to channels, so the flag is just enabling and disabling the metric? Should we enable/disable the whole thing based on the configuration setting for monitoring?

Reply:
There is already a setting to turn of shared channels, in which case the registration will fail. IMO that is enough should we want to disable it.  However, that means OnActivate will fail and the plugin will not run.  I think we need to decide if disabled shared channels is a fatal plugin error.  Today, probably not a fatal error, but perhaps in the future.   We could also look at the plugin enabling shared channels as needed.

Reply:
Checking the code, registration will succeed regardless of whether shared channels service is enabled.  If disabled you simply won't get any sync messages.

Reply:
> shared channels service is enabled

Ah, this is interesting @wiggin77. Can you elaborate on what's required to enable? Does enabling force the user to enable the full shared channels experience, or can we selectively enable this just for plugins?